vm/systemSmartContracts/staking.go

```diff
@@ -1289,6 +1311,12 @@ func (s *stakingSC) isNodeJailedOrWithBadRating(registrationData *StakedDataV2_0
 }
 
 func (s *stakingSC) getWaitingListIndex(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if s.flagStakingV4.IsSet() {
+		s.eei.AddReturnMessage(vm.ErrWaitingListDisabled.Error())
+		s.eei.Finish([]byte{0})
+
+		return vmcommon.Ok
```

Comment:
if you have return message than return vmcommon.UserError, and do not add to Finish anything. as it is ignored

Reply:
Thanks for heads up, refactored it